### ISSUE TITLE
[Phobos Bug] Units are already dead but still in map

### ISSUE THREAD

**Issue by Zorbung** [Phobos-developers/Phobos#307]
### Description:
This isn't 100% fixed, there are still games that don't end when short games is disabled. Soviet AI players sometimes remain alive even if there are no more buildings/units, I suspect it could be a drone that hasn't been deleted properly (probably when the parasited unit has been killed but that's just an assumption) because sometimes when selecting all drones on the map (with T) I get a drone that's outside the map.

### Phobos Build:
0.2.1.1

### Conditions:
Rare bug for Soviet players, but happens more often if there are many soviet players on the map.

### Steps To Reproduce:
1. Play as soviet on a close quarters map where fighting happens a lot.
2. Increasing drone's speed from rulesmd.ini may accelerate/cause the bug (just an assumption).
3. Keep making drones and attacking with them.
4. Whenever you only have 1 drone, select all with T and order selection to move.
5. Pressing Z (waypoint) will show movement to target lines; when bugged you will get an extra line that comes from outside the map (usually northside of the map).
6. If point 5 fails and all opponents look defeated, you may still get to see a soviet player that hasn't died yet when pressing the "diplomacy" button.

### Expected Behaviour:
Games should correctly end when a soviet players has no more units and buildings.

### Actual Behaviour:
The bugged soviet player remains alive and the game doesn't end. Other AI can't target it either, although sometimes AI may fire superweapons on a dead location trying to kill it, but there's nothing at that location.

**Comment by Metadorius** [comment]
Does this happen without Phobos? It seems to me like a vanilla bug.

**Comment by Zorbung** [comment]
Yes, it happens without Phobos too. Since this was presumably fixed in a previous release I marked it as a Phobos bug to notify that the fix is incomplete.

**Comment by Metadorius** [comment]
The fix you mentioned has nothing to do with this one, so it's a vanilla bug. Perhaps the expression wasn't clear enough though.

**Comment by Zorbung** [comment]
This seems to happen a lot to Cuba house, it may also be related to Terrorists.
